iotedgedev/connectionstring.py

```diff
@@ -13,6 +13,8 @@ def __init__(self, value):
 
             if self.data:
                 self.HostName = self["hostname"]
+                if self.HostName:
```

Comment:
Keep `def __init__(self, value)` as is.

Change self.value to self.ConnectionString

`self.ConnectionString = value`

And then update all references to self.value to self.ConnectionString.

Also, line 7 above references value.split(';'), that should be self.ConnectionString.split(';')